BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -112,6 +125,34 @@ public virtual void Initialize(TelemetryConfiguration configuration, TelemetryMo
             this.Initialize(configuration, modules, PlatformSingleton.Current.ReadConfigurationXml());
         }
 
+        [SuppressMessage("Microsoft.Reliability", "CA2000:DisposeObjectsBeforeLosingScope", Justification = "We want objects created in this method to live for the life of the application.")]
+        [SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", Justification = "This value is not used in NetStandard1.3 but is used for all other frameworks..")]
+        internal void EvaluateSelfDiagnosticsMode(TelemetryModules modules)
+        {
+#if !NETSTANDARD1_3
+            if (PlatformSingleton.Current.TryGetEnvironmentVariable(SelfDiagnosticsEnvironmentVariable, out string selfDiagnosticsConfigurationString))
+            {
+                var keyValuePairs = SelfDiagnosticsProvider.ParseConfigurationString(selfDiagnosticsConfigurationString);
+
+                if (SelfDiagnosticsProvider.IsFileDiagnostics(keyValuePairs, out string path, out string level, out string maxSize))
+                {
```

Comment:
where is maxSize used?

Reply:
at the moment it's not. :(
it was identified in the spec, java specifically wants to use it. but the library we're using doesn't provide us that kind of control over the output file.

I was torn between parsing it so we could use it later, or ignoring it in our SDK.

Reply:
i removed the maxSize since we can't support it at this time